### PR TITLE
Adding a C64 Mini Keyboard product to Keyboards

### PR DESCRIPTION
**Pull request by bleughb**
Submitting a new c64 Mini keyboard matrix. This is for a Keyboard kit sold by Myself to upgrade the Small Commodore 64 Mini device to have a fully working keyboard.

## Description

No Changes to main code, just the addition of an extra keyboard product firmware / map

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/c64/C64.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    K00, K01, K02, K03, K04, K05, K06, K07, K08,                                                                 \
+    K10, K11, K12, K13, K14, K15, K16, K17, K18,                                                                \
+    K20, K21, K22, K23, K24, K25, K26, K27, K28,                                                                \
+    K30, K31, K32, K33, K34, K35, K36, K37, K38,                                                            \
+    K40, K41, K42, K43, K44, K45, K46, K47, K48,                                                            \
+    K50, K51, K52, K53, K54, K55, K56, K57,                                                             \
+    K60, K61, K62, K63, K64, K65, K66, K67,                                                             \
+    K70, K71, K72, K73, K74, K75, K76, K77,                                                             \
+    K80, K81, K82, K83, K84, K85                                                                           \
+) { \
```
Comment (drashna) [inline keyboards/c64/C64.h:29]: This should be reformatted to match the layout of the keyboard, not the matrix. 

You can look at boards like the planck ez, gmmk pro and the like to see what I mean.

File: keyboards/c64/config.h [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x6060
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Bleugh
+#define PRODUCT         C64_Mini_Keyboard
+#define DESCRIPTION     USB Keyboard for the C64 Mini by Retrogames
```
Comment (drashna) [inline keyboards/c64/config.h:31]:
```suggestion
```

File: keyboards/c64/config.h [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+/* Keyboard Firmware for TheC64mini by retrogames. This firmware works on the Upgrade Keyboard Kit sold by Dean Woodyatt at Bleugh.biz */
+#ifndef CONFIG_H
+#define CONFIG_H
```
Comment (drashna) [inline keyboards/c64/config.h:21]:
```suggestion
#pragma once
```

File: keyboards/c64/config.h [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+#define MATRIX_ROW_PINS { B6, B2, B3, B1, F7, F6, F5, F4, D3 }		// PROBABLY FULLY WORKING PINS 
+
+#define UNUSED_PINS
+// #define BACKLIGHT_PIN B7
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION ROW2COL
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCING_DELAY 5
```
Comment (drashna) [inline keyboards/c64/config.h:58]:
```suggestion
#define DEBOUNCE 5
```

File: keyboards/c64/config.h [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+//#define MAGIC_KEY_LAYER6         6
+//#define MAGIC_KEY_LAYER7         7
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+#define MAGIC_KEY_BOOTLOADER       B
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (drashna) [inline keyboards/c64/config.h:144]:
```suggestion
```

File: keyboards/c64/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,128 @@
+#include "C64.h"
```
Comment (drashna) [inline keyboards/c64/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/c64/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+// Fillers to make layering more clear
+#define _______ KC_TRNS
+#define ___T___ KC_TRNS
+#define XXXXXXX KC_NO
```
Comment (drashna) [inline keyboards/c64/keymaps/default/keymap.c:6]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/c64/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+
+// Layer shorthand
+#define _QW 	0
+#define _ALT 	1
+#define _SHFT 	2
```
Comment (drashna) [inline keyboards/c64/keymaps/default/keymap.c:11]:
```suggestion
enum custom_layers {
    _QW,
    _ALT,
    _SHFT,
};
```

File: keyboards/c64/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+            unregister_code(KC_RSFT);
+          }
+        case 3: // this would trigger when you hit a key mapped as M(3)
+          if (record->event.pressed) {
+			return MACRO( I(255), D(LALT), T(P1), T(P5), T(P6), U(LALT), END );
+          }
+          break;
+      }
+    return MACRO_NONE;
+};
```
Comment (drashna) [inline keyboards/c64/keymaps/default/keymap.c:128]:
```suggestion
```

File: keyboards/c64/rules.mk [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/c64/rules.mk:11]:
```suggestion
```

File: keyboards/c64/rules.mk [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
+
+LAYOUTS = LAYOUT_C64Mini
+
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/c64/rules.mk:35]:
```suggestion
BOOTMAGIC_ENABLE = yes       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes        # Mouse keys
EXTRAKEY_ENABLE = yes        # Audio control and System control
CONSOLE_ENABLE = yes         # Console for debug
COMMAND_ENABLE = yes         # Commands for debug and configuration
NKRO_ENABLE = no             # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no        # Enable keyboard backlight functionality
AUDIO_ENABLE = no            # Audio output
UNICODE_ENABLE = no          # Unicode
BLUETOOTH_ENABLE = no        # Enable Bluetooth
RGBLIGHT_ENABLE = no         # Enable WS2812 RGB underlight.

# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend

LTO_ENABLE = yes
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by bleughb** [comment]
HI, I deleted the old request fully, and have submitted a new one with i hope all the correct changes,
thankyou for your patience, i'm new at this :-)

**Pull request closed without merging**